### PR TITLE
MAINT: Remove wminkowski

### PR DESCRIPTION
**Pull request by h-vetinari**
Since @peterbell10's #12374, `wminkowski` has been scheduled for removal in 1.8 - let's do that.

~While I was add it, I saw a lot of loops over `_METRICS_NAMES` in `test_distance.py` that lent themselves well to parametrization.~ will appear in follow-up PR

I recommend to review the diff per commit (ideally) and ignoring whitespace changes (use ⚙️ in the diff view)

### CODE REVIEW

**Comment by h-vetinari** [comment]
Strictly speaking, everything after the first commit could be omitted for 1.8, if time is short. I still think it's a good clean-up though...

Will also need a news entry.

File: scipy/spatial/conftest.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import pytest
+
+from scipy.spatial.distance import _METRICS_NAMES
+
+
+@pytest.fixture(params=_METRICS_NAMES)
+def metric(request):
```
Comment (peterbell10) [inline scipy/spatial/conftest.py:7]: Can you move the test changes to another PR and consider just using `pytest.mark.parametrize` instead of a parametrized fixture?

**Comment by h-vetinari** [comment]
> Can you move the test changes to another PR

I removed the commits for the parametrization, will post them in another PR after this one is merged.

> consider just using `pytest.mark.parametrize` instead of a parametrized fixture?

I'd suggest we discuss this in that PR then.

**Comment by h-vetinari** [comment]
> Will also need a news entry.

Seems that all of this gets auto-generated for the release? At least I couldn't find where I'd have to add release notes for this.

**Comment by rgommers** [comment]
> Seems that all of this gets auto-generated for the release? At least I couldn't find where I'd have to add release notes for this.

https://github.com/scipy/scipy/wiki/Release-note-entries-for-SciPy-1.8.0

**Comment by h-vetinari** [comment]
> https://github.com/scipy/scipy/wiki/Release-note-entries-for-SciPy-1.8.0

Thanks; I wasn't aware of this and had been looking in-repo. I'll add a description once this PR gets merged - anything left to do here?

Proposed text for the release notes (presumably under "Backwards incompatible changes"):
> The function ``scipy.spatial.distance.wminkowski`` has been removed, after having been documented as deprecated since scipy 1.0, and raising a warning since scipy 1.3. To achieve the same results as before, please use the `minkowski` distance function with the (optional) ``w=`` keyword-argument for the given weight.

**Comment by rgommers** [comment]
> Proposed text for the release notes (presumably under "Backwards incompatible changes"):

That's the right place. Text sounds fine, I'd shorten it a bit and leave out _"after having been documented as deprecated since scipy 1.0, and raising a warning since scipy 1.3."_

**Comment by h-vetinari** [comment]
> Text sounds fine, I'd shorten it a bit and leave out

Sure, no problem

**Comment by h-vetinari** [comment]
@rgommers 
If there are no further blockers, would you mind merging this? There are two other PRs stacked on top of this (one not yet submitted), and if at all reasonable and possible, I wanted to get #15055 into 1.8 because it's pretty trivial.

**Review by rgommers — APPROVED** [review]
All LGTM, thanks @h-vetinari
